docs/course-best-practices.md

```diff
@@ -0,0 +1,519 @@
+# Course management best practices
```

Comment:
TODO: Change title to reflect that this is for large courses with multiple developers.